### PR TITLE
Support Relative URL for Image src

### PR DESCRIPTION
**Pull request by nateevans**
see https://github.com/fabricjs/fabric.js/issues/5476

### CODE REVIEW

File: src/shapes/image.class.js [hunk 1]
```diff
@@ -349,7 +357,13 @@
         if (element.toDataURL) {
           return element.toDataURL();
         }
-        return element.src;
+
+        if (this.srcFromAttribute && typeof element.getAttribute === 'function') {
```
Comment (asturur) [inline src/shapes/image.class.js]: We support ie10 ( sort of ) ie11 and then modern browsers.
Can you verify if getAttribute is really unsupported from some of those?

**Comment by asturur** [comment]
@nateevans a couple of comments:
Please branch of your master from where you are, restore your master to the top of fabric.js master, and before committing the code remember to do:

`git checkout master dist` so that you do not commit changes to the dist folder.

Can you try to add a test in the relative file in the test/unit folder?

File: dist/fabric.js [hunk 2]
```diff
@@ -19479,6 +19479,14 @@ fabric.util.object.extend(fabric.Object.prototype, /** @lends fabric.Object.prot
      */
     strokeWidth: 0,
 
+    /**
+     * When calling {@link fabric.Image.getSrc}, return value from element src with `element.getAttribute('src')` (if available).
+     * This allows for relative urls as image src.
+     * @type Boolean
+     * @default
```
Comment (asturur) [inline dist/fabric.js]: can you add:
@since 2.7.0 ?

File: src/shapes/image.class.js (hunk 1 above)
Comment (asturur) [reply src/shapes/image.class.js]:
```suggestion
        if (this.srcFromAttribute) {
```

**Comment by asturur** [comment]
@nateevans a couple of comments:
You should not commit to the dist folder.
In order to roll it back either

branch off your master from where you are, restore your master to the top of fabric.js master, and before committing the code remember to do:
`git checkout master dist` so that you do not commit changes to the dist folder.

or find the commit where dist was not changed and do
`git checkout ..commithash... dist`

**Comment by asturur** [comment]
i wonder how we can properly test node. as it is now the test under node does not make much sense.

**Comment by nateevans** [comment]
@asturur 
> You should not commit to the dist folder.

Ah ok. How do you run tests normally? seems like I had to build those in order to run `testem` since it runs off of `dist/`

> i wonder how we can properly test node. as it is now the test under node does not make much sense.

I don't think the use case for this actually applies to node. With JSDOM in Node, calling `element.src` doesn't seem to add any schema/hostname. Which makes sense, I don't know what it would even add there... it just seems to affect usage in browser.

**Comment by nateevans** [comment]
I'll close out this PR and make one from the new branch

**Pull request closed without merging**